### PR TITLE
procedurally generate background based on signer key

### PR DESCRIPTION
**Pull request by jacogr**
https://github.com/ethcore/parity/issues/2228

Next up -

- as part of setup, user can have some way of selecting (signer token setup with initial token)
- allow changing of the background in settings (signer keys stays the same, postfix for key changes)
- ~~need to adjust Modals (look slightly out of place)~~
- ~~ParityBar (dapp overlay) needs to get the same styling as modals (at least container overlays)~~

Caveats -

- currently with each refresh a new background will be used. This is on purpose (adding the current timestamp as part of the token sha3) - it allows us to go through as many backgrounds as possible and see how things fit together from a UI perspective.

In action -

parity 2016-09-22 00-57-15
parity 2016-09-22 00-56-52
parity 2016-09-22 00-56-32
parity 2016-09-22 00-56-15
parity 2016-09-22 00-55-56
parity 2016-09-22 00-55-38
parity 2016-09-22 00-55-26
parity 2016-09-22 00-55-08
parity 2016-09-22 00-54-53
parity 2016-09-22 00-54-40
parity 2016-09-22 00-54-22

### CODE REVIEW

File: js/src/ui/ProcBackground/procBackground.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import GeoPattern from 'geopattern';
+import React, { Component, PropTypes } from 'react';
+
+export default class ProcBackground extends Component {
```
Comment (derhuerst) [inline js/src/ui/ProcBackground/procBackground.js]: I'm not happy with the name `ProcBackground`. I can't tell what `Proc` means if I don't know what it's about. I'd prefer e.g. `GeneratedBackground`, `PatternBackground` or – similar to `IdentityIcon` – `IdentityBackground`.

File: js/src/ui/ProcBackground/procBackground.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      return;
+    }
+
+    this.updateBackground(newProps.seed);
+  }
+
+  render () {
+    const { children } = this.props;
+    const { backgroundImage } = this.state;
+    const style = { backgroundImage, width: '100%', height: '100%' };
```
Comment (derhuerst) [inline js/src/ui/ProcBackground/procBackground.js]: Setting `width` and `height` to `100%` may be dangerous in some places? Are they required?

File: js/src/ui/Theme/theme.js [hunk 3]
```diff
@@ -28,11 +28,11 @@ muiTheme.raisedButton.primaryTextColor = 'white';
 muiTheme.snackbar.backgroundColor = 'rgba(255, 30, 30, 0.9)';
 muiTheme.snackbar.textColor = 'rgba(255, 255, 255, 0.9)';
 muiTheme.tabs = lightTheme.tabs;
-muiTheme.tabs.backgroundColor = 'rgb(65, 65, 65)';
+muiTheme.tabs.backgroundColor = 'none'; // 'rgba(65, 65, 65, 0.75)';
```
Comment (derhuerst) [inline js/src/ui/Theme/theme.js]: Just taste, but I think it looks better without transparency. 😄

File: js/src/ui/ProcBackground/procBackground.js (hunk 1 above)
Comment (jacogr) [inline js/src/ui/ProcBackground/procBackground.js]: I have renamed it - agreed with you on that actually, it is now ParityBackground, exactly what the function is.

File: js/src/ui/ProcBackground/procBackground.js (hunk 2 above)
Comment (jacogr) [inline js/src/ui/ProcBackground/procBackground.js]: Sadly, yes. With the height we only stretch to child height, which means the image may just cover parts of the screen. Other option may be to set the child heights as/when required.

File: js/src/ui/Theme/theme.js (hunk 3 above)
Comment (jacogr) [inline js/src/ui/Theme/theme.js]: I think we need to get it in mostly as-in and see what bugs us. There will need to be tweaks to make everything fit and look optimal.

File: js/src/ui/ProcBackground/procBackground.js (hunk 2 above)
Comment (tomusdrw) [inline js/src/ui/ProcBackground/procBackground.js]: Why not: `position: absolute; top: 0; bottom: 0; left: 0; right: 0;`?
```
<div>
  <div style={style}></div> 
  { children }
</div>
```
Could be even a separate `ReactDOM.render` call.

Comment (jacogr) [inline js/src/ui/ProcBackground/procBackground.js]: I have removed that. The height was wrong (re-usable components), so if needed (which is one place), the className can set it. Actually in that place Application/Container the className already does.

File: js/src/ui/Theme/theme.js (hunk 3 above)
Comment (jacogr) [inline js/src/ui/Theme/theme.js]: Adjusted it slightly. There is method behind the madness, with some transparency (0.7 now, up from 0.5) the background colors bleed through. The trick is to get it just enough so it bleeds enough, but not too much that it is distracting.

As of now, quite happy as an itial stab. Will need to tweak as things look out of place, but for that need testing. (Plus, meeting with UX guy on Monday, we may get some help from a pure styling perspective, so not going to overthink it.)

**Pull request merged**